python/lsst/ctrl/bps/wms/panda/cmd_line_embedder.py

```diff
@@ -0,0 +1,94 @@
+import os
+
+
+class CommandLineEmbedder:
+    """
+    Class embeds static (constant across a task) values into the pipeline execution command line
+    and resolves submission side environment variables
+
+    Parameters
+    ----------
+    config : `~lsst.ctrl.bps.BPSConfig`
+            BPS configuration that includes the list of dynamic (uniques per job)
+            and submission side resolved variables
+
+    """
+    def __init__(self, config):
+        self.leave_placeholder_params = config.get("placeholderParams", ['qgraphNodeId', 'qgraphId'])
+        self.submit_side_resolved = config.get("submitSideResolvedParams", ['USER'])
+
+    def replace_static_parameters(self, cmd_line, lazy_vars):
+        """ Substitutes the lazy parameters in the command line which are static,
+        the same for every job in the workflow and could be defined once. This function offloads the edge
+        node processing and number of parameters transferred together with job
+
+
+        Parameters
+        ----------
+        cmd_line: `str` command line to be processed
+        lazy_vars: `dict` of lazy variables and its values
+
+        Returns
+        -------
+        processed command line
+        """
+
+        for param_name, param_val in lazy_vars.items():
+            if param_name not in self.leave_placeholder_params:
+                cmd_line = cmd_line.replace('{'+param_name+'}', param_val)

```

Comment:
Not sure that it will affect the cmd_line or not, but in the bps yaml, those "{value}" can actually contain f-string formats (e.g., {value:02}).

Reply:
If we are using f-string style logic here we should not be using simple string replace but instead should be calling `str.format()` and you can pass in a dict to that directly.